src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -122,13 +122,14 @@ void AsyncOpenTask::attach_to_subscription_initializer(AsyncOpenCallback&& async
                                                        std::shared_ptr<_impl::RealmCoordinator> coordinator,
                                                        bool rerun_on_launch)
 {
-    // Subscription initialization will occure in either of these 2 cases.
-    // 1. The realm file has been created (this is the first time we ever open the file). The latest
-    //    subscription version must be 0 when we open realm for the first time, and then become 1 as soon as the
-    //    subscription is committed. This will happen when the subscription initializer callback is invoked by
-    //    coordinator->get_realm().
-    // 2. we are instructed to run the subscription initializer via sync_config->rerun_init_subscription_on_open.
-    //    But we do that only if this is the first time we open realm.
+    // Attaching the subscription initializer to the latest subscription that was committed.
+    // This is going to be enough for covering:
+    // 1) the first subscription (version 0) has been committed and the database was opened for the first time
```

Comment:
The comment is not 100% accurate. The two cases, the way I read the code, are:

1. The last subscription set committed is not Complete yet. This was either committed in a previous session or in the current session (either by the subscription initializer or the user)
2. The database was opened for the first time and rerun on lunch was set to true.

Also, if `sub_state` is Complete, the second part of the condition has no effect (i.e, get_state_change_notification returns immediately). In other words, the first part is enough (but as mentioned before, it changes the previous behavior)